docs/reference/changelog.md

```diff
@@ -179,45 +177,70 @@ List.uniqueBy_v0 [1;2;3;4] (fun x -> Int.divide_v0 x 2) = [ 1, 3, 4 ] // old Dar
-messages have changed. If you were relying on the explicit format of a Dark
-language or stdlib error message, expect that it will be different.
+### Generating JSON
 
-If you do any error handling which relies specifically on the text of an error
-message coming from Dark, we recommend you no longer do that, and just use the
-presence of the error instead of the text.
+We have changed how JSON is generated in many cases. All the JSON generated by
+Dark is now standards compliant, and uses a different formatting.
 
-There are two places in Dark which use string error messages:
+In the old version of Dark, we would generate invalid JSON for the Float values
+`NaN`, `Infinity` and `-Infinity`. The old version of Dark generates them as
+bare identifiers, while the new version puts them in a string (eg `NaN` vs
+`"NaN"`).
 
-#### `Result.Error`s
+This may occur in any of the places in which we generate JSON, which are:
 
-The `Error` enum (referred to as `Result.Error` below for clarity), will contain
-a string error in most cases, which you might be using directly or indirectly.
+- when responding to a HTTP request in the HTTP framework
+- when making a request with HttpClient (any version)
+- when calling any of:
+  - `Dict::toJson_v0`
+  - `Object::toJson_v1` (_deprecated_)
+  - `Object::toJson_v0` (_deprecated_)
+  - `Twilio::sendText_v1`
+  - `Twilio::sendText_v0` (_deprecated_)
 
-You might be accessing `Result.Error`s and their text contents directly using
-the `match` statement.
+Note that the following `JWT` functions do not use this new behaviour, and
+should have the exact same behaviour as before, including:
 
-You might also be accessing their text contents indirectly, using `toString`, or
-other stringifying functions, such as `toString`, `JSON::` functions and
-`HTTPClient::` functions. This is only true if you taken the function returning
-the `Result.Error` off the error rail. This text may also make it to your
-web/mobile clients or API consumers.
+- `JWT::signAndEncode_v0` (_deprecated_)
+- `JWT::signAndEncode_v1`
+- `JWT::signAndEncodeWithHeaders_v0` (_deprecated_)
+- `JWT::signAndEncodeWithHeaders_v1`
 
-`Result.Error`s returned via a HTTP handler are not presented to the user, so
-text cannot leak that way.
+### Parsing JSON
 
-#### Runtime errors
+When parsing JSON, we no longer accept the following invalid JSON:
 
-Runtime errors (including type errors) are not accessible via Dark programs, and
-will always terminate the Dark program when they are accessed. As a result, the
-text of any runtime error should not be accessible to your programs or your
-users.
+- bare field names, such as in `{ id : 5 }`. You need to quote field names:
+  `{ "id": 5 }`
+- newlines and invalid bytes in JSON strings
+- the bare identifiers `NaN`, `Infinity` and `-Infinity` are no longer parsed
+  into valid floats (note that the old version of Dark might have generated JSON
+  with these values in it)
```

Comment:
Can/should we avail a fn to users which turns such invalid JSON into valid JSON?

Reply:
At this point everyone has been switched over, so I don't think anyone would have use for one.